test/fixtures/test-runner/custom_reporters/error_reporter.mjs

```diff
@@ -0,0 +1,25 @@
+export default async function* customReporter(source) {
+  for await (const event of source) {
+    switch (event.type) {
+      case 'test:start':
+        throw new Error("Custom reporter error")
+      case 'test:pass':
+        yield `test ${event.data.name} passed\n`;
+        break;
+      case 'test:fail':
+        yield `test ${event.data.name} failed\n`;
+        break;
+      case 'test:plan':
+        yield 'test plan';
+        break;
+      case 'test:diagnostic':
+        yield `${event.data.message}\n`;
+        break;
+      case 'test:coverage': {
+        const { totalLineCount } = event.data.summary.totals;
+        yield `total line count: ${totalLineCount}\n`;
+        break;
```

Comment:
Do we actually want/need all these?
I think best to keep things as focused at what they test as possible 🙂

Reply:
No, you're right, removed